bluetoe/link_layer/include/bluetoe/peripheral_latency.hpp

```diff
@@ -0,0 +1,318 @@
+    {
+        /**
+         * @brief listen at all connection events.
+         *
+         * This does not imply, that the link layer will also respond at every
+         * connection event.
+         */
+        listen_always,
+
+        /**
+         * @brief listen, if the link layer contains unacknowlaged data.
+         *
+         * If a PDU was send out, but no acknowlagement was jet received,
+         * the link layer will listen at the next connection event, if
+         * this options is set.
+         *
+         * This might be interesting, if the sending queue is very small because
+         * Bluetoe has to keep a copy of a transmitted PDU, as long as that PDU
+         * is not acknowlaged.
+         */
+        listen_if_unacknowlaged_data,
+
+        /**
+         * @brief listen to the next connection event, if the last received PDU was
+         *        not empty.
+         */
+        listen_if_last_received_not_empty,
+
+        /**
+         * @brief listen to the next connection event, if the last transmitted PDU was
+         *        not empty.
+         */
+        listen_if_last_transmitted_not_empty,
+
+        /**
+         * @brief listen to the next connection event, if the last received PDU had
+         *        the more data (MD) flag set.
+         *
+         * With the MD flag, the central (and the peripheral) can indicate, that there
+         * is more data to be send. Usually this data is then transmitted during a
+         * connection even. If there is no support for the MD flag either by the peripheral
+         * or by the central, then, this option might be usefull.
+         */
+        listen_if_last_received_had_more_data,
+
+        /**
+         * @brief listen to the next connection event, if for the last connection event,
+         *        a crc error happend, or a timeout occured.
+         *
+         * This behavior is required by the Bluetooth specification and should not left out
+         * without good reason.
+         */
+        listen_if_last_received_had_error,
+
+        /**
+         * @brief always respond to a first PDU from the central at a connection event.
+         *
+         * If the link layer listened to the connection event and received a valid PDU
+         * from the central, it will reply.
+         */
+        send_always,
+
+        /**
+         * @brief if there is pending data to be send out, the stack will send that
+         *        data at the next scheduled connection event.
+         *
+         * If the stack is configured to use peripheral latency and decided to leave out
+         * a few connection events, and then data to be send out becomed available, the
+         * next scheduled event will not be changed. If the desired behaviour is, to send
+         * the data out with the very next connection event, this option has to be used
+         * in conjunction with `listen_always`.
+         */
+        send_pending_data,
```

Comment:
Why do we need to listen always for this to work, aren't we missing an option 'listen_if_pending_tx' ?

Reply:
This was a proposal. I already realized, that this will most likely not work as described there.